### PR TITLE
fix: [CCM-5377]: fix clone perspective not cloning rules

### PR DESCRIPTION
**Pull request by rajarshee-harness**
##### Summary:

Using a new endpoint for cloning a perspective instead of using the one for creating a new perspective.

##### Jira Links:

https://harness.atlassian.net/browse/CCM-5377

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by rajarshee-harness** [comment]
trigger rebase

**Comment by rajarshee-harness** [comment]
trigger cypress

File: src/modules/75-ce/components/PerspectiveViews/__tests__/mockPerspectiveData.json [hunk 1]
```diff
@@ -0,0 +1,2228 @@
+[
+  {
```
Comment (jenil-harness) [inline src/modules/75-ce/components/PerspectiveViews/__tests__/mockPerspectiveData.json:2]: Too big data for mock
Maybe keep fewer items that are exclusive.
Refrain from using original ids and account ids.

File: src/modules/75-ce/pages/perspective-list/PerspectiveListPage.tsx [hunk 2]
```diff
@@ -540,9 +580,9 @@ const PerspectiveListPage: React.FC = () => {
                 featureNames: [FeatureIdentifier.PERSPECTIVES]
               }
             }}
-            onClick={async () => {
+            onClick={() => {
               trackEvent(USER_JOURNEY_EVENTS.CREATE_NEW_PERSPECTIVE, {})
-              await createNewPerspective({}, false)
+              createNewPerspective({})
```
Comment (jenil-harness) [inline src/modules/75-ce/pages/perspective-list/PerspectiveListPage.tsx:585]: Removed this?
Does create perspective button shows loader while the api is in progess?

File: src/modules/75-ce/pages/perspective-list/PerspectiveListPage.tsx [hunk 3]
```diff
@@ -385,22 +386,27 @@ const PerspectiveListPage: React.FC = () => {
     }
   })
 
+  const { mutate: cloneView } = useClonePerspective({
+    perspectiveId: '', // this will be set by cloneView fn when called.
```
[the remark this answers is not in the document]
Comment (rajarshee-harness) [reply src/modules/75-ce/pages/perspective-list/PerspectiveListPage.tsx:390]: Yes, it was required in `UseClonePerspectiveProps`

File: src/modules/75-ce/pages/perspective-list/PerspectiveListPage.tsx (hunk 2 above)
Comment (rajarshee-harness) [reply src/modules/75-ce/pages/perspective-list/PerspectiveListPage.tsx:585]: Yes, also it was showing `'await' has no effect on the type of this expression.` and sonar was showing a critical code smell.

**Comment by rajarshee-harness** [comment]
trigger rebase

**Comment by rajarshee-harness** [comment]
trigger rebase

File: src/modules/75-ce/components/PerspectiveViews/PerspectiveGridView.tsx [hunk 4]
```diff
@@ -89,7 +90,7 @@ const PerpsectiveGridCard: (props: PerspectiveGridCardProps) => JSX.Element | nu
   }
 
   const onCloneClick: () => void = () => {
-    clonePerspective(data, true)
+    data?.id && data?.name && clonePerspective(data.id, data.name)
```
Comment (rishik30) [inline src/modules/75-ce/components/PerspectiveViews/PerspectiveGridView.tsx:93]: change to `if` syntax

File: src/modules/75-ce/components/PerspectiveViews/PerspectiveListView.tsx [hunk 5]
```diff
@@ -160,7 +161,7 @@ const PerspectiveListView: React.FC<PerspectiveListViewProps> = ({
     const onCloneClick: (e: any) => void = e => {
       e.stopPropagation()
       setMenuOpen(false)
-      clonePerspective(row.original, true)
+      row.original.id && row.original.name && clonePerspective(row.original.id, row.original.name)
```
Comment (rishik30) [inline src/modules/75-ce/components/PerspectiveViews/PerspectiveListView.tsx:164]: change to `if` syntax

File: src/modules/75-ce/components/PerspectiveViews/__tests__/PerspectiveGridView.test.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+import React from 'react'
+import { render } from '@testing-library/react'
+import { TestWrapper } from '@common/utils/testUtils'
+
+import PerspectiveGridView from '@ce/components/PerspectiveViews/PerspectiveGridView'
+import type { QlceView } from 'services/ce/services'
+import mockPerspectiveData from './mockPerspectiveData.json'
+
+describe('test cases for multiple conatiner header', () => {
```
Comment (rishik30) [inline src/modules/75-ce/components/PerspectiveViews/__tests__/PerspectiveGridView.test.tsx:16]: include button click test cases as well

File: src/modules/75-ce/pages/perspective-list/PerspectiveListPage.tsx [hunk 7]
```diff
@@ -421,6 +427,40 @@ const PerspectiveListPage: React.FC = () => {
...
+        history.push(
+          routes.toCECreatePerspective({
+            accountId: accountId,
+            perspectiveId: uuid
+          })
+        )
+      }
+    } catch (e: any) {
+      const errMessage = e.data.message
+      showError(errMessage)
```
Comment (rishik30) [inline src/modules/75-ce/pages/perspective-list/PerspectiveListPage.tsx:460]: use `getErrorInfoFromErrorObject(e)` instead

**Comment by rajarshee-harness** [comment]
trigger rebase

**Comment by rajarshee-harness** [comment]
retrigger jest

**Comment by rajarshee-harness** [comment]
retrigger jest

**Review by jenil-harness — APPROVED** [review]

**Review by rishik30 — APPROVED** [review]

**Review by vkbansal — APPROVED** [review]

**Pull request merged**